src/Cli.Tests/ConfigureOptionsTests.cs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Cli.Tests
+{
+    /// <summary>
+    /// Validates that `dab configure [options]` configure settings with the user-provided options.
+    /// </summary>
+    [TestClass]
+    public class ConfigureOptionsTests : VerifyBase
+    {
+        private MockFileSystem? _fileSystem;
+        private FileSystemRuntimeConfigLoader? _runtimeConfigLoader;
+        private const string TEST_RUNTIME_CONFIG_FILE = "test-update-runtime-setting.json";
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            _fileSystem = FileSystemUtils.ProvisionMockFileSystem();
+
+            _runtimeConfigLoader = new FileSystemRuntimeConfigLoader(_fileSystem);
+
+            ILoggerFactory loggerFactory = TestLoggerSupport.ProvisionLoggerFactory();
+
+            SetLoggerForCliConfigGenerator(loggerFactory.CreateLogger<ConfigGenerator>());
+            SetCliUtilsLogger(loggerFactory.CreateLogger<Utils>());
+        }
+
+        /// <summary>
+        /// Validates that `dab configure` is a no-op because the user provided no options to update
+        /// the config file. The config file must not change.
```

Comment:
> The config file must not change.

This is valid when the config has `{"max-depth": null}` or `{"max-depth": -1}` . For any other valid integer [1,Int.max], the config will change

Reply:
thanks, updated.